src/components/Vector3Input.js

```diff
@@ -7,11 +7,11 @@ export default function Vector3Input({ value, onChange, ...rest }) {
   return (
     <div className={styles.vector3Input}>
       <div className={styles.label}>X:</div>
-      <NumericInput value={value.x} onChange={x => onChange("x", x)} {...rest} />
+      <NumericInput value={value.x} onChange={x => onChange({ x, y: value.y, z: value.z })} {...rest} />
```

Comment:
Had the `Vector3Input` send the full state since it would be difficult for the receiver to reconstruct the rest of the state otherwise.